torchqdynamics/utils/utils.py

```diff
@@ -173,70 +171,126 @@ def dissipator(L: Tensor, rho: Tensor) -> Tensor:
 def lindbladian(H: Tensor, Ls: Tensor, rho: Tensor) -> Tensor:
     r"""Apply the Lindbladian superoperator to a density matrix.
 
-    The system Lindbladian $\mathcal{L}(\cdot)$ is the superoperator
-    generating the evolution of the system. It is defined by
+    The Lindbladian superoperator $\mathcal{L}$ is defined by:
+
     $$
-        \frac{\mathrm{d}\rho}{\mathrm{d}t} = \mathcal{L}(\rho) = -i[H,\rho]
-                                             + \sum_{i=1}^n \mathcal{D}[L_i](\rho).
+        \mathcal{L}(\rho) = -i[H,\rho] + \sum_{k=1}^N \mathcal{D}[L_k](\rho),
     $$
 
+    where $H$ is the system Hamiltonian, $\{L_k\}$ is a set of $N$ jump operators
+    (arbitrary operators) and $\mathcal{D}[L]$ is the Lindblad dissipation superoperator
+    (see [dissipator()][torchqdynamics.dissipator]).
+
+    Note:
+        This superoperator is also sometimes called the *Liouvillian*.
+
     Args:
-        H: Hamiltonian of dimension `(..., d, d)`.
-        Ls: Jump operators tensor of dimension `(..., n, d, d)`.
-        rho: Density matrix of dimension `(..., d, d)`.
+        H (..., n, n): Hamiltonian.
+        Ls (..., N, n, n): Sequence of jump operators (arbitrary operators).
+        rho (..., n, n): Density matrix.
 
     Returns:
-        Tensor of dimension `(..., d, d)` resulting from the application of the
-        Lindbladian.
+        (..., n, n): Resulting operator (it is not a density matrix).
     """
     return -1j * (H @ rho - rho @ H) + dissipator(Ls, rho).sum(0)
 
 
-def tensprod(*x: Tensor) -> Tensor:
-    """Compute the tensor product of a sequence of state vectors, density matrices or
+def tensprod(*args: Tensor) -> Tensor:
+    r"""Tensor product of a sequence of state vectors, density matrices or
     operators.
 
+    Examples:
+        >>> psi = tq.tensprod(
+        ...     tq.coherent(20, 2.0),
+        ...     tq.fock(2, 0),
+        ...     tq.fock(5, 1)
+        ... )
+        >>> psi.shape
+        torch.Size([200, 1])
+
+        >>> rho = tq.tensprod(
+        ...     tq.coherent_dm(20, 2.0),
+        ...     tq.fock_dm(2, 0),
+        ...     tq.fock_dm(5, 1)
+        ... )
+        >>> rho.shape
+        torch.Size([200, 200])
+
+    The returned tensor shape is:
+
+    - $(n, 1)$ with $n=\prod_k n_k$ if all input tensors are state vectors with shape
+      $(n_k, 1)$,
+    - $(n, n)$ with $n=\prod_k n_k$ if all input tensors are density matrices or
+      operators vectors with shape $(n_k, n_k)$.
+
+    Warning:
+        This function does not yet support bra.
+
+    Warning:
+        This function does not yet support arbitrarily batched tensors (see
+        [issue #69](https://github.com/pierreguilmin/torchqdynamics/issues/69)).
+
     Note:
-        This function is the equivalent of `qutip.tensor`.
-    """
-    return reduce(torch.kron, x)
+        This function is the equivalent of `qutip.tensor()`.
 
+    Args:
+        *args (n_k, 1) or (n_k, n_k): Sequence of state vectors, density matrices or
```

Comment:
```suggestion
        *args (n, 1) or (n, n): Sequence of state vectors, density matrices or
```
I know you wanted to match the above notations, but it makes it harder to read IMO.

Reply:
I agree it's a bit harder to read, but I think it's crucial to keep it, to differ from a function which would require same shape tensors (as I would expect when reading the proposed suggestion).